docs/INTEGRATION.md

```diff
@@ -0,0 +1,789 @@
+| `NoSchedule`        | Fatal errors requiring remediation | New pods without toleration won't be scheduled |
+| `PreferNoSchedule`  | Degraded state or warnings       | Scheduler tries to avoid but will schedule if necessary |
+| `NoExecute`         | Immediate evacuation needed      | Existing pods without toleration are evicted (rarely used) |
+
+### Configuring Taints
+
+Taints are defined in Fault Quarantine rulesets. Here's an example showing how to enable taints:
+
+```yaml
+# distros/kubernetes/nvsentinel/charts/fault-quarantine/values.yaml
+rulesets:
+  - version: "1"
+    name: "GPU XID Critical Errors"
+    priority: 100
+    match:
+      any:
+        - kind: "HealthEvent"
+          expression: 'event.checkName == "GpuXidError" && event.isFatal == true'
+    # Uncomment to enable tainting:
+    taint:
+      key: "nvidia.com/gpu-xid-error"  # Choose your own key format
+      value: "true"                    # Or use "fatal", "degraded", etc.
+      effect: "NoSchedule"
+    cordon:
+      shouldCordon: true  # Enabled by default
+```
+
+**Key Points:**
+- Taints are **commented out by default** - you must enable them
+- You control the taint key format (`nvidia.com/*` or `gpu.health/*` or any custom format)
+- You control the taint values (`true`, `fatal`, `degraded`, etc.)
+- Cordoning is enabled by default; tainting is opt-in
+
+### Integration Patterns
+
+**Check if node has any NVIDIA-related taints:**
+```shell
+kubectl get nodes -o json | jq '.items[] 
+  | select(.spec.taints[]? 
+  | select(.key | startswith("nvidia.com/"))) 
+  | .metadata.name'
+```
+
+**Check for specific error type:**
+```shell
+kubectl get nodes -o json | jq '.items[] 
+  | select(.spec.taints[]? 
+  | select(.key == "nvidia.com/gpu-xid-error")) 
+  | .metadata.name'
+```
+
+**Tolerate specific taints in pod specs:**
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: gpu-workload
+spec:
+  tolerations:
+    # Match the exact taint configured in your rulesets
+    - key: "nvidia.com/gpu-xid-error"
+      operator: "Equal"
+      value: "true"
+      effect: "NoSchedule"
+```
+
+**Watch for taint changes (automation):**
+
+```go
+informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+    UpdateFunc: func(oldObj, newObj interface{}) {
+        newNode := newObj.(*corev1.Node)
+        
+        // Check for NVIDIA-related taints
+        for _, taint := range newNode.Spec.Taints {
+            if strings.HasPrefix(taint.Key, "nvidia.com/") {
+                // Trigger alert, update scheduler, etc.
+                log.Printf("Node %s has taint %s=%s", 
+                    newNode.Name, taint.Key, taint.Value)
+            }
+        }
+    },
+})
+```
+
+---
+
+## 2. Why is a Node Bad? Check Node Conditions
+
+**Use node conditions for monitoring, alerting, and detailed diagnostics.**
+
+While taints tell you "this node is bad", conditions tell you *why* it's bad. Use conditions for dashboards, alerts, and troubleshooting.
```

Comment:
Can we also recommend using kube-state-metrics and provide some specific examples of the metrics to use?